<launch>
  <!-- Arguments -->
  <arg name="use_camera" default="true" />
  <arg name="use_video_file" default="false" />
  <arg name="video_path" default="" />
  <arg name="camera_topic" default="/zed2i/zed_node/rgb/image_rect_color" />
  <arg name="camera_info_topic" default="/zed2i/zed_node/rgb/camera_info" />

  <!-- Start SLAM node -->
  <node pkg="autonomy" type="slam.py" name="monocular_slam" output="screen">
    <param name="use_camera" value="$(arg use_camera)" />
    <param name="use_video_file" value="$(arg use_video_file)" />
    <param name="video_path" value="$(arg video_path)" />
    <param name="camera_topic" value="$(arg camera_topic)" />
    <param name="camera_info_topic" value="$(arg camera_info_topic)" />
  </node>

  <!-- Optionally launch RViz for visualization -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find autonomy)/config/slam_visualization.rviz" if="$(arg use_camera)" />
</launch>
